crates/core/app/tests/mock_consensus.rs

```diff
@@ -78,71 +82,40 @@ async fn mock_consensus_can_send_a_spend_action() -> anyhow::Result<()> {
 
     // Sync the mock client, using the test account's full viewing key, to the latest snapshot.
-    let MockClient { notes, sct, .. } = MockClient::new(test_keys::FULL_VIEWING_KEY.clone())
+    let (viewing_key, spend_key) = (&test_keys::FULL_VIEWING_KEY, &test_keys::SPEND_KEY);
+    let client = MockClient::new(test_keys::FULL_VIEWING_KEY.clone())
         .with_sync_to_storage(&storage)
         .await?;
+    let client_note_count_pre_spend = client.notes.len();
 
     // Take one of the test account's notes...
-    let note = notes
-        .values()
-        .cloned()
+    let commitment = client
+        .notes
+        .keys()
+        .copied()
         .next()
-        .ok_or_else(|| anyhow!("mock client had no note"))?;
-    let asset_id = note.asset_id();
-    let proof = sct
-        .witness(note.commit())
-        .ok_or_else(|| anyhow!("index is not witnessed"))?;
-
-    // ...and use it to craft a `Spend`.
-    let (spend, spend_key) = {
-        use {decaf377_rdsa::SigningKey, penumbra_shielded_pool::SpendPlan};
-        let spend_plan = SpendPlan::new(&mut rng, note, proof.position());
-        let auth_sig = test_keys::SPEND_KEY
-            .spend_auth_key()
-            .randomize(&spend_plan.randomizer)
-            .sign(&mut rng, [0u8; 64].as_ref());
-        let spend = spend_plan.spend(&test_keys::FULL_VIEWING_KEY, auth_sig, proof, sct.root());
-        let key = SigningKey::from(spend_plan.value_blinding);
-        (spend, key)
-    };
-
-    // Next, craft a transaction, containing this `Spend`.
-    let tx = {
-        use {
-            penumbra_asset::Value,
-            penumbra_fee::Fee,
-            penumbra_num::Amount,
-            penumbra_transaction::{Action, Transaction, TransactionBody, TransactionParameters},
-            penumbra_txhash::AuthorizingData,
-        };
-        let transaction_parameters = TransactionParameters {
-            expiry_height: 0,
-            chain_id: "i-wonder-if-this-is-load-bearing".to_owned(),
-            fee: Fee(Value {
-                amount: Amount::zero(),
-                asset_id,
-            }),
-        };
-        let transaction_body = TransactionBody {
-            actions: vec![Action::Spend(spend)],
-            transaction_parameters,
-            ..Default::default()
-        };
-        let binding_sig = spend_key.sign(rng, transaction_body.auth_hash().as_bytes());
-        let transaction = Transaction {
-            transaction_body,
-            binding_sig,
-            anchor: sct.root(),
-        };
-        <Transaction as penumbra_proto::DomainType>::encode_to_vec(&transaction)
-    };
+        .ok_or_else(|| anyhow!("mock client had no note"))?
+        .tap(|note| tracing::info!(?note, "mock client note commitment"));
+
+    // XXX(kate): this should probably move into the mock-client library, rather than living here
+    // as an extension trait. for now, this is defined here to avoid dealing with dependencies.
```

Comment:
I think it'd be better to leave the plan creation inline here, and later evaluate how to use it with the `Planner` API.

I don't think we want to create a general-purpose `spend_tx` method on the `MockClient`; it's not possible to make a general purpose API like that and starting down that path will lead us to another half-baked implementation of a `Planner` that's only used for tests.

Reply:
sounds good. the latest commit 81eea7755 moves this inline.